### DISCUSSION TITLE
Crazyfile 2.0, attempted to update to 2023.02 firmware, now it doesn't do anything

### DISCUSSION THREAD

**Discussion by robacarp** [bitcraze/discussions#632] [Q&A] [answered]
I have an old Crazyfile 2.0 I just found in an old box, and I thought it'd be fun to tinker with again. I charged it up and had it running, controlling it with the ios app "Crazyfile". Then I decided to update the firmware, which seemed to go well until right at the end. The iOS app then said "Bootloader Error:" (nothing after the colon). I pressed OK, and it then indicated that it had succeeded at updating the firmware. 

I powered off the Crazyfile, back on, and tried to get it connected again to the iOS app, but it won't do anything. I can tell that it is connecting because when I power off the drone, the app informs me the connection has been terminated.

I think I can tell that the app has been un-published at this point, so perhaps I need to try some other method of flashing. Following the documentation floating around the internet is a dizzying tale of dead ends, EoL forums, and out of date wiki's. My use case is dead simple though, I just want the stock firmware and a remote of some sort to get it running.

What should I do next?

**Comment by ataffanel** [comment, marked as answer]
Hi,

We have retired the forum and the wiki in favor of documentation in repos displayed on the [main website](https://www.bitcraze.io/documentation/repository/) and GitHub discussion (here :).

I am not sure what you mean about the app being unpublished though, as far as I can see it is still available in the app store.

Though, I could verify the the bootloader client in iOS is indeed broken: I see the same error as you when I try to flash my Crazyflie with it. I created an issue in bitcraze/crazyflie2-ios-client#41, this needs to be investigated further. In the mean-time you will need a Crazyradio and a PC to flash your Crazyflie.

It is not possible to brick the Crazyflies from the bootloader so it is always possible to start it again. In your case the nRF51 firmware (that handles the button) is still functional so you can just switch on the Crazyflie by pressing the button until the blue LED blinks. Even if the nRF51 firmware was damaged it would still be possible to start the bootloader by removing the battery, pressing the button, and connecting the battery while the button is pressed.

Sorry about that, the iOS bootloader failure was not detected until now. We will have a look at it, if you are interested in iOS development do not hesitate to debug it yourself.

**Reply by robacarp** [reply]
> I am not sure what you mean about the app being unpublished though, as far as I can see it is still available in the app store.

Good to know -- when I searched for it, I found nothing and had to go back to "My Purchases" to get it again. I feared that it'd been made obsolete somehow.

I'm happy to dive into the Crazyradio/PC flashing for now. I don't have any experience doing iOS development. Do you have a link to a reasonably recently documented process for flashing with Crazyradio?

**Reply by ataffanel** [reply]
The [getting started guide](https://www.bitcraze.io/documentation/tutorials/getting-started-with-crazyflie-2-x/) is a good place to start. I points to instructions to [install the client](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/installation/install/) and [flash the latest version of the firmware](https://bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/#firmware-upgrade).
